### ISSUE TITLE
[ENH] Postprocessing Methods for Probabilistic Forecasts

### ISSUE THREAD

**Issue by benHeid** [sktime/sktime#6558]
**Is your feature request related to a problem? Please describe.**
When doing quantile based forecasts, the quantiles might be created that is not optimal regarding a specific metric. E.g. consider the following example below. 

```python
from sktime.datasets import load_forecastingdata
from sktime.transformations.series.difference import Differencer
from sktime.transformations.series.detrend import Deseasonalizer
from sktime.split import temporal_train_test_split
from sktime.forecasting.conformal import ConformalIntervals
import pandas as pd
from sktime.forecasting.naive import NaiveForecaster

def australian_electricity_demand_dataset_t2(series="T2"):
    data = load_forecastingdata("australian_electricity_demand_dataset")[0]
    data = data.set_index("series_name")
    series = data.loc[series]["series_value"]
    return pd.DataFrame(series, index=pd.date_range("2006-01-01", periods=len(series), freq="30min"))[-30*48:]

y = australian_electricity_demand_dataset_t2()
y_train, y_test = temporal_train_test_split(y, test_size=48)

forecaster = Differencer(lags=[1]) * Deseasonalizer(sp=48) * ConformalIntervals(forecaster=NaiveForecaster(sp=48), 
                        method="conformal",
                        sample_frac=0.005)
forecaster.fit(y_train, fh=y_test.index
               )

from sktime.performance_metrics.forecasting.probabilistic import CRPS, PinballLoss



res = forecaster.predict(fh=y_test.index)

res_int = forecaster.predict_interval(coverage=[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95])

qs = []
for q in [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95]: 
    _res_int = res_int[[(0, q, "lower"),(0, q, "upper")]]
    _res_int.columns = pd.MultiIndex.from_tuples([(0, 0.9, "lower"),(0, 0.9, "upper")])
    qs.append(PinballLoss().evaluate(y_test, _res_int))

pd.Series(qs, index=[0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 0.95]).plot.bar()

```

The resulting plot looks like: 

<img width="630" alt="image" src="https://github.com/sktime/sktime/assets/9725296/ed2c0320-bb33-4aee-a0f7-758b9a6cc1e8">

Showing the Pinball loss on the y-axis and the alpha  of the forecaster' intervals on the x-axis. These intervals are renamed so that the PinballLoss is considering them always as 0.9 interval. Now, we observe that the smallest PinballLoss considering the 0.9 interval for alpha 0.4. 

**Describe the solution you'd like**
I would like to have postprocessing methods, that transform the probabilistic forecast in a way that it minimizes the error regarding a specific metric.

**Comment by fkiraly** [comment]
Fully agreed! This would be an important roadmap item for the probabilistic project. This is perhaps the same issue on `skpro`, at a slightly higher level of abstractions: https://github.com/sktime/skpro/issues/321 - I think we ended up wanting to schedule this at one of the dev meetings. We need to resolve the constraint that you are not available at te Monday slots.

Issues related to this in `skpro`:

* the extension of the transformations design to probabilistic outputs https://github.com/sktime/skpro/issues/321
* transformed distribution API https://github.com/sktime/skpro/issues/30
* probabilistic `TransformedTargetRegressor` https://github.com/sktime/skpro/issues/31
* multi-index support, required for panel/hierarchical: https://github.com/sktime/skpro/issues/212
